### PR TITLE
Refactor task scheduler and executor

### PR DESCRIPTION
**Pull request by yah01**
Make the performance able to scale out

Signed-off-by: yah01 <yang.cen@zilliz.com>

/kind improvement
related #20769

### CODE REVIEW

File: configs/milvus.yaml [hunk 1]
```diff
@@ -184,6 +184,7 @@ queryCoord:
   loadTimeoutSeconds: 600
   checkHandoffInterval: 5000
   taskMergeCap: 16
+  queryCoord.taskExecutionCap: 256
```
Comment (congqixia) [inline configs/milvus.yaml:187]: `queryCoord.taskExecutionCap` is already under `queryCoord` section.

File: internal/querycoordv2/task/scheduler.go [hunk 2]
```diff
@@ -527,7 +547,28 @@ func (scheduler *taskScheduler) process(task Task) bool {
...
+	for step < len(actions) && actions[step].IsFinished(scheduler.distMgr) {
+		task.StepUp()
+		step++
+	}
+
+	if step == len(actions) {
+		step--
+	}
+
+	executor, ok := scheduler.executors[actions[step].Node()]
```
Comment (xiaofan-luan) [inline internal/querycoordv2/task/scheduler.go:560]: what will happen if the task executor has been removed?
wait for task timeout?

Comment (yah01) [reply internal/querycoordv2/task/scheduler.go:560]: all tasks related to the node will be stale, and then removed, `checkStale()` checks whether the node is offline per scheduling round

**Pull request merged**